devtools/dump_devinfo.py

```diff
@@ -64,11 +65,34 @@ def default_to_regular(d):
 @click.command()
 @click.argument("host")
 @click.option("-d", "--debug", is_flag=True)
-def cli(host, debug):
+@click.option(
+    "-t",
+    "--type",
+    envvar="KASA_TYPE",
+    default="smartdevice",
+    type=click.Choice(list(TYPE_TO_CLASS), case_sensitive=False),
+)
+@click.option(
+    "--username",
+    default=None,
+    required=False,
+    envvar="TPLINK_CLOUD_USERNAME",
+    help="Username/email address to authenticate to device.",
+)
+@click.option(
+    "--password",
+    default=None,
+    required=False,
+    envvar="TPLINK_CLOUD_PASSWORD",
+    help="Password to use to authenticate to device.",
+)
+def cli(host, debug, type, username, password):
     """Generate devinfo file for given device."""
     if debug:
         logging.basicConfig(level=logging.DEBUG)
 
+    dev = TYPE_TO_CLASS[type](host, credentials=Credentials(username, password))
```

Comment:
`--type` defaults to `smartdevice` which should maintain the previous behavior since it's protocol will be `TPLinkSmartHomeProtocol`

Reply:
So I spotted this I wanted to use it to test if all the devices I have respond correctly to the `{} instead of null` queries yesterday, but it's erroring out:
`Error: Invalid value for '-t' / '--type': 'smartdevice' is not one of 'plug', 'bulb', 'dimmer', 'strip', 'kasacam', 'lightstrip'.` :-)

Reply:
Removed